desc/magnetic_fields.py

```diff
@@ -1846,6 +1853,328 @@ def from_surface(
+        )
+
+    def change_resolution(
+        self,
+        L_B=None,
+        M_B=None,
+        L_x=None,
+        M_x=None,
+        N_x=None,
+        NFP=None,
+    ):
+        """Set the spectral resolution of well and shift terms.
+
+        Parameters
+        ----------
+        L_B : int
+            Radial resolution of the magnetic well parameters B_lm.
+        M_B : int
+            Number of spline points in the magnetic well parameters B_lm.
+        L_x : int
+            Radial resolution of x_lmn.
+        M_x : int
+            Poloidal resolution of x_lmn.
+        N_x : int
+            Toroidal resolution of x_lmn.
+        NFP : int
+            Number of field periods.
+
+        """
+        old_L_B = self.L_B
+
+        self._NFP = setdefault(NFP, self.NFP)
+        self._L_B = setdefault(L_B, self.L_B)
+        self._M_B = setdefault(M_B, self.M_B)
+        self._L_x = setdefault(L_x, self.L_x)
+        self._M_x = setdefault(M_x, self.M_x)
+        self._N_x = setdefault(N_x, self.N_x)
+
+        # change well parameters and basis
+        rho = (  # Chebyshev-Gauss-Lobatto nodes
+            1 - np.cos(np.arange(old_L_B // 2, old_L_B + 1, 1) * np.pi / old_L_B)
+        ) / 2
+        nodes = np.array([rho, np.zeros_like(rho), np.zeros_like(rho)]).T
+
+        transform_fwd = self.B_basis.evaluate(nodes)
+        transform_rev = scipy.linalg.pinv(transform_fwd)
+        B_old = transform_fwd @ self.B_lm.reshape((old_L_B + 1, -1))
+
+        eta_old = np.linspace(0, jnp.pi / 2, num=B_old.shape[-1])
+        eta_new = np.linspace(0, jnp.pi / 2, num=self.M_B)
+
+        B_new = np.zeros((old_L_B + 1, self.M_B))
+        for i in range(old_L_B + 1):
+            B_new[i, :] = interp1d(eta_new, eta_old, B_old[i, :], method="monotonic-0")
+        B_lm_old = transform_rev @ B_new
+
+        old_modes_well = self.B_basis.modes
+        self.B_basis.change_resolution(self.L_B)
+
+        B_lm_new = np.zeros((self.L_B + 1, self.M_B))
+        for j in range(self.M_B):
+            B_lm_new[:, j] = copy_coeffs(
+                B_lm_old[:, j], old_modes_well, self.B_basis.modes
+            )
+        self._B_lm = B_lm_new.flatten()
+
+        # change shift parameters and basis
+        old_modes_shift = self.x_basis.modes
+        self.x_basis.change_resolution(
+            self.L_x, self.M_x, self.N_x, NFP=self.NFP, sym="cos(t)"
+        )
+        self._x_lmn = copy_coeffs(self.x_lmn, old_modes_shift, self.x_basis.modes)
+
+    def compute(
+        self,
+        names,
+        grid=None,
+        params=None,
+        transforms=None,
+        profiles=None,
+        data=None,
+        **kwargs,
+    ):
+        """Compute the quantity given by name on grid.
+
+        Parameters
+        ----------
+        names : str or array-like of str
+            Name(s) of the quantity(s) to compute.
+        grid : Grid, optional
+            Grid of coordinates to evaluate at. Defaults to the quadrature grid.
```

Comment:
Should specify what coordinates is this grid referring to (rho, eta, alpha right?)

Reply:
i.e. is theta = eta and zeta = alpha?